chain/vm/vm.go

```diff
@@ -607,16 +607,30 @@ func (vm *VM) transfer(from, to address.Address, amt types.BigInt) aerrors.Actor
 		return nil
 	}
 
+	fromId, err := vm.cstate.LookupID(from)
+	if err != nil {
+		return aerrors.Fatalf("transfer failed when resolving sender address: %s", err)
+	}
+
+	toId, err := vm.cstate.LookupID(to)
+	if err != nil {
+		return aerrors.Fatalf("transfer failed when resolving receiver address: %s", err)
+	}
+
+	if fromId == toId {
+		return nil
```

Comment:
Will this fail when there are not enough funds to transfer? I guess yes because there are some message checks earlier right?

Reply:
Yeah, that should be caught at the beginning of `ApplyMessage`...Might be worth adding a redundant check anyway for internal messages.